data-prepper-plugins/sqs-source/src/main/java/org/opensearch/dataprepper/plugins/source/sqs/QueueConfig.java

```diff
@@ -0,0 +1,91 @@
+package org.opensearch.dataprepper.plugins.source.sqs;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Max;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import java.time.Duration;
+
+import org.hibernate.validator.constraints.time.DurationMax;
+import org.hibernate.validator.constraints.time.DurationMin;
+
+public class QueueConfig {
+
+    private static final Integer DEFAULT_MAXIMUM_MESSAGES = null;
+    private static final Boolean DEFAULT_VISIBILITY_DUPLICATE_PROTECTION = false;
+    private static final Duration DEFAULT_VISIBILITY_TIMEOUT_SECONDS = null;
+    private static final Duration DEFAULT_VISIBILITY_DUPLICATE_PROTECTION_TIMEOUT = Duration.ofHours(2);
+    private static final Duration DEFAULT_WAIT_TIME_SECONDS = Duration.ofSeconds(20);
+    private static final Duration DEFAULT_POLL_DELAY_SECONDS = Duration.ofSeconds(0);
+    static final int DEFAULT_NUMBER_OF_WORKERS = 1;
+
+    @JsonProperty("url")
+    @NotNull
+    private String url;
+
+    @JsonProperty("workers") 
+    @Valid
+    private int numWorkers = DEFAULT_NUMBER_OF_WORKERS;
+
+    @JsonProperty("maximum_messages")
+    @Min(1)
+    @Max(10)
+    private Integer maximumMessages = DEFAULT_MAXIMUM_MESSAGES;
+
+    @JsonProperty("poll_delay")
+    @DurationMin(seconds = 0)
+    private Duration pollDelay = DEFAULT_POLL_DELAY_SECONDS;
+
+    @JsonProperty("visibility_timeout")
+    @DurationMin(seconds = 0)
+    @DurationMax(seconds = 43200)
+    private Duration visibilityTimeout = DEFAULT_VISIBILITY_TIMEOUT_SECONDS;
+
+    @JsonProperty("visibility_duplication_protection")
+    private Boolean visibilityDuplicateProtection = DEFAULT_VISIBILITY_DUPLICATE_PROTECTION;
```

Comment:
There is no `@NotNull` annotation here. Default for boxed type Boolean is null. Just want to make sure null is an expected value here. Otherwise, just change it to a primitive boolean type.